core/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -49,15 +51,22 @@
  * @param mergingOptions are the options related to the subsequence match merging mechanism that opposed obfuscation.
  * @param normalize enables additional normalization mechanisms. Only supported by some language modules.
  * @param analyzeComments If true, comments will be extracted from the submissions.
+ * @param isFrequencyAnalysisEnabled if isFrequencyAnalysisEnabled Analysis is used
+ * @param frequencyAnalysisStrategy strategy for determining the isFrequencyAnalysisEnabled
+ * @param frequencyStrategyMinValue min considered subsequence length in frequencyStrategies
+ * @param weightingStrategy weighting function used in the isFrequencyAnalysisEnabled Analysis
+ * @param weightingStrategyWeightingFactor factor how strong the considered influence of the weighting function
+ * (maximal) can be
  */
 @RecordBuilder()
 public record JPlagOptions(@JsonSerialize(using = LanguageSerializer.class) Language language, Integer minimumTokenMatch,
         @JsonSerialize(contentUsing = FileSerializer.class) Set<File> submissionDirectories,
         @JsonSerialize(contentUsing = FileSerializer.class) Set<File> oldSubmissionDirectories,
         @JsonSerialize(using = FileSerializer.class) File baseCodeSubmissionDirectory, String subdirectoryName, List<String> fileSuffixes,
         String exclusionFileName, SimilarityMetric similarityMetric, double similarityThreshold, int maximumNumberOfComparisons,
-        ClusteringOptions clusteringOptions, boolean debugParser, MergingOptions mergingOptions, boolean normalize, boolean analyzeComments)
-        implements JPlagOptionsBuilder.With {
+        ClusteringOptions clusteringOptions, boolean debugParser, MergingOptions mergingOptions, boolean normalize, boolean analyzeComments,
+        boolean isFrequencyAnalysisEnabled, FrequencyAnalysisStrategy frequencyAnalysisStrategy, int frequencyStrategyMinValue,
+        MatchFrequencyWeightingFunction weightingStrategy, double weightingStrategyWeightingFactor) implements JPlagOptionsBuilder.With {
```

Comment:
Correction for comment above (use .With also for your withX implementation in the other record :)